packages/@aws-cdk/cdk/lib/physical-name.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Represents the human-readable, physical name of a resource.
+ * These can be known at code runtime,
+ * in the case of fixed, customer-provided names,
+ * at synthesis time, in case of automatically assigned names by the framework,
+ * or only at deploy time, assigned by CloudFormation -
+ * which is the default.
+ *
+ * @see #deployTime()
+ * @see #deployTimeOrAssigned()
+ * @see #fixed()
+ */
+export abstract class PhysicalName {
```

Comment:
I am wondering if it makes sense to provide an option for `synthTime` (e.g. if I want my resource names to be always be automatically generated at synthesis time). We can leave this for later though.

Reply:
I've thought about it, but I couldn't come up with a scenario when someone would want that behavior. Like you said, I'm fairly confident we'll be able to provide it later if we find a compelling use case/customer demand for this functionality.